hledger-lib/Hledger/Data/Posting.hs

```diff
@@ -246,25 +253,23 @@ relatedPostings _ = []
 
 -- | Does this posting fall within the given date span ?
 isPostingInDateSpan :: DateSpan -> Posting -> Bool
-isPostingInDateSpan s = spanContainsDate s . postingDate
+isPostingInDateSpan = isPostingInDateSpan' PrimaryDate
 
 -- --date2-sensitive version, separate for now to avoid disturbing multiBalanceReport.
 isPostingInDateSpan' :: WhichDate -> DateSpan -> Posting -> Bool
-isPostingInDateSpan' PrimaryDate   s = spanContainsDate s . postingDate
-isPostingInDateSpan' SecondaryDate s = spanContainsDate s . postingDate2
+isPostingInDateSpan' w s = spanContainsDate s . postingDate' w
```

Comment:
I'm ok with this if you think it's better, but I do think that secondary is a misfeature and I'd be in favour of not letting it spread even further through the code.

Reply:
I'm happy to get rid of that. Out of curiosity, what would your vision be for getting rid of SecondaryDate?

Reply:
Since it's working and helps Ledger compatibility I think we should keep it. But I'd be glad to keep it's impact to the minimum.